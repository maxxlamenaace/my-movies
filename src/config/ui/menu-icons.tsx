import React, { ReactElement } from 'react';

import HomeOutlinedIcon from '@mui/icons-material/HomeOutlined';
import SlideshowOutlinedIcon from '@mui/icons-material/SlideshowOutlined';
import LiveTvOutlinedIcon from '@mui/icons-material/LiveTvOutlined';
import FavoriteBorderOutlinedIcon from '@mui/icons-material/FavoriteBorderOutlined';
import SearchOutlinedIcon from '@mui/icons-material/SearchOutlined';

export type MenuIcon = {
  display: 'home' | 'movies' | 'series' | 'search' | 'favorites';
  path: string;
  icon: ReactElement;
};

export const menuIcons: MenuIcon[] = [
  {
    display: 'home',
    path: '/',
    icon: <HomeOutlinedIcon />,
  },
  {
    display: 'movies',
    path: '/medias/movie',
    icon: <SlideshowOutlinedIcon />,
  },
  {
    display: 'series',
    path: '/medias/tv',
    icon: <LiveTvOutlinedIcon />,
  },
  {
    display: 'search',
    path: '/search',
    icon: <SearchOutlinedIcon />,
  },
  {
    display: 'favorites',
    path: '/favorites',
    icon: <FavoriteBorderOutlinedIcon />,
  },
];
