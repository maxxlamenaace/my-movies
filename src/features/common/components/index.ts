import Loader from './loader/Loader';
import Logo from './logo/Logo';
import Footer from './footer/Footer';
import AppBar from './app-bar/AppBar';
import Container from './container/Container';
import ScrollableAppBar from './scrollable-app-bar/ScrollableAppBar';
import Sidebar from './sidebar/Sidebar';
import ManualSwiper from './manual-swiper/ManualSwiper';
import Wrapper from './wrapper/Wrapper';
import PaginatorSwiper from './paginator-swiper/PaginatorSwiper';

export {
  Loader,
  Logo,
  Footer,
  AppBar,
  Container,
  ScrollableAppBar,
  Sidebar,
  ManualSwiper,
  Wrapper,
  PaginatorSwiper,
};
