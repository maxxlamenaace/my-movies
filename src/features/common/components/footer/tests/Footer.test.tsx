import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import { cleanup, render } from '@testing-library/react';

import { Footer } from '@/features/common';
import { menuIcons } from '@/config/ui/menu-icons';

describe('Footer', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const { asFragment } = render(<Footer />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with as many buttons menus as needed', () => {
    const { container } = render(<Footer />, { wrapper: BrowserRouter });
    expect(container.querySelectorAll('#footer-menu-button')).toHaveLength(menuIcons.length);
  });

  it('renders with a logo', () => {
    const { container } = render(<Footer />, { wrapper: BrowserRouter });
    expect(container.querySelector('#logo')).toBeDefined();
  });
});
