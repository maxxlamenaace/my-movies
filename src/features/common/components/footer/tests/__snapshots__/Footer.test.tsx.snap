// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1cig8sb"
  >
    <div
      class="css-678bp8-MuiStack-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 css-p7vaay-MuiPaper-root"
      >
        <div
          class="css-1n12vvt-MuiStack-root"
        >
          <a
            href="/"
            id="logo-link"
            style="text-decoration: none;"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-10bvdjj-MuiTypography-root"
              id="logo"
              style="color: rgba(0, 0, 0, 0.87);"
            >
              my
              <span
                style="color: rgb(25, 118, 210);"
              >
                Movies
              </span>
            </p>
          </a>
          <div
            class="MuiBox-root css-0"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14tur9a-MuiButtonBase-root-MuiButton-root"
              href="/"
              id="footer-menu-button"
              tabindex="0"
            >
              home
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14tur9a-MuiButtonBase-root-MuiButton-root"
              href="/medias/movie"
              id="footer-menu-button"
              tabindex="0"
            >
              movies
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14tur9a-MuiButtonBase-root-MuiButton-root"
              href="/medias/tv"
              id="footer-menu-button"
              tabindex="0"
            >
              series
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14tur9a-MuiButtonBase-root-MuiButton-root"
              href="/search"
              id="footer-menu-button"
              tabindex="0"
            >
              search
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-14tur9a-MuiButtonBase-root-MuiButton-root"
              href="/favorites"
              id="footer-menu-button"
              tabindex="0"
            >
              favorites
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
